### PR TITLE
Update readme and installation docs

### LINKED ISSUE #1712: Update README.md
### Which components does the task require to be changed? (think hard pls)

README.md

### Description

- [ ] Update the gif so it is up to date
- [ ] Mark coverage as wip or done
- [ ] Update installation guide, explicitly mention rust requirement in there

### LINKED ISSUE #2690: Prioritize asdf installation over using the script
### Which component is your bug related to?

snforge

### Foundry Version

0.33.0

### What operating system are you using?

Linux

### What system architecture are you using?

x86

### What happened

Update the docs to list asdf as the preferred way of installation, and add the instructions to run `asdf install starknet-foundry latest`

### Trace

_No response_

### Is there an existing issue for this?

- [ ] I have searched the existing issues and verified no issue exits for this problem.

### LINKED ISSUE #2721: Remove cairo-lang-sierra dependency

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1712
Closes #2690 
Closes #2721

## Introduced changes

- Updated installation instructions in readme and docs
- Updated GIF in docs
- Removed "Features" section as it duplicates "Roadmap"

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by franciszekjob — COMMENTED** [review]
Did you verify it by following these instructions on each platform?

File: docs/src/getting-started/installation.md [hunk 1]
```diff
@@ -15,53 +19,56 @@ all installed and added to your `PATH` environment variable.
...
-You can install it by running:
+First, add the Starknet Foundry plugin to asdf:
 
 ```shell
-$ curl -L https://raw.githubusercontent.com/foundry-rs/starknet-foundry/master/scripts/install.sh | sh
+$ asdf plugin add starknet-foundry
 ```
 
-Follow the instructions and then run:
+##### Common Error
```
Comment (Arcticae) [inline docs/src/getting-started/installation.md:35]: I lack other examples here, namely ones with rust <1.84

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -74,7 +171,7 @@ Starknet Foundry is under active development! Expect a lot of new features to ap
 - [x] Deployment scripts written in Cairo
 - [ ] Transactions profiling 🏗️
 - [ ] Debugging utilities 🏗️
-- [ ] Test coverage reports (check out [cairo-coverage](https://github.com/software-mansion/cairo-coverage)) 🏗️ 
+- [ ] Test coverage reports (check out [cairo-coverage](https://github.com/software-mansion/cairo-coverage)) 🏗️
```
Comment (ddoktorski) [inline README.md:174]: Maybe let's add a point about parameterized tests to the roadmap?

File: docs/src/getting-started/installation.md [hunk 3]
```diff
@@ -15,53 +19,56 @@ all installed and added to your `PATH` environment variable.
...
+
+Users may encounter this error when trying to use `snforge` or `sncast` without setting a version:
 
 ```shell
-$ snfoundryup
+No version is set for command snforge
+Consider adding one of the following versions in your config file at starknet-foundry 0.32.0
```
Comment (ddoktorski) [inline docs/src/getting-started/installation.md:41]:
```suggestion
Consider adding one of the following versions in your config file at $HOME/.tool_versions
starknet-foundry 0.32.0
```

File: README.md [hunk 4]
```diff
@@ -32,31 +32,128 @@ Starknet Foundry, like its [Ethereum counterpart](https://github.com/foundry-rs/
...
+
+To verify that ASDF was correctly installed, run
+
+```shell
+asdf --version
+```
+
+#### Install Scarb
+
+First, add Scarb plugin to ASDF
```
Comment (kkawula) [inline README.md:63]:
```suggestion
First, add Scarb plugin to asdf
```

File: README.md [hunk 5]
```diff
@@ -32,31 +32,128 @@ Starknet Foundry, like its [Ethereum counterpart](https://github.com/foundry-rs/
...
+2. Extract it to a location where you would like to have Scarb installed. We recommend `%LOCALAPPDATA%\Programs\scarb`.
+3. From this directory, get the full path to `scarb\bin` and add it to PATH.
+   See [this article](https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/) for instructions on
+   Windows 10 and 11.
+
+To verify that Scarb was correctly installed, run
+
+```shell
+scarb --version
+```
```
Comment (kkawula) [inline README.md:134]: Maybe add link to scarb docs?  https://docs.swmansion.com/scarb/download.html#windows

File: README.md [hunk 6]
```diff
@@ -32,31 +32,128 @@ Starknet Foundry, like its [Ethereum counterpart](https://github.com/foundry-rs/
...
+
+Install Scarb
 
 ```shell
-$ curl -L https://raw.githubusercontent.com/foundry-rs/starknet-foundry/master/scripts/install.sh | sh
+asdf install scarb latest
 ```
 
-Follow the instructions and then run
+To verify that Scarb was correctly installed, run
```
[the remark this answers is not in the document]
Comment (cptartur) [reply README.md:75]: Changed to `To verify that Scarb was installed, run`.

File: docs/src/getting-started/installation.md (hunk 1 above)
Comment (cptartur) [reply docs/src/getting-started/installation.md:35]: Added

File: README.md (hunk 2 above)
Comment (cptartur) [reply README.md:205]: Is this feature big enough to include it here?

Comment (ddoktorski) [reply README.md:205]: Your call

**Review by ddoktorski — APPROVED** [review]

File: docs/src/getting-started/installation.md [hunk 7]
```diff
@@ -90,14 +80,65 @@ As for now, Starknet Foundry on Windows needs manual installation, but necessary
...
+For additional information on asdf version management, see
+the [asdf](https://asdf-vm.com/guide/getting-started.html#_6-set-a-version)
+
+### Invalid Rust Version
+
+When running any `snforge` command, error similar to this is displayed
+
+```shell
+Compiling snforge_scarb_plugin v0.34.0
+error: package `snforge_scarb_plugin v0.34.0 cannot be built because it requires rustc 1.80.1 or newer, while the currently active rustc version is 1.76.0
```
Comment (franciszekjob) [inline docs/src/getting-started/installation.md:118]: Looks like typo 😅 
```suggestion
error: package snforge_scarb_plugin v0.34.0 cannot be built because it requires rustc 1.80.1 or newer, while the currently active rustc version is 1.76.0
```

**Review by franciszekjob — APPROVED** [review]

**Comment by cptartur** [comment]
I've moved the installation entirely to docs and just linked it in readme. I think it makes more sense to not duplicate it & it encourages the users to actually checkout our docs.

**Review by ddoktorski — APPROVED** [review]

File: docs/src/getting-started/installation.md [hunk 8]
```diff
@@ -61,36 +216,50 @@ This error indicates that `Starknet Foundry` version is unset. To resolve it, se
...
-2. Extract it to a location where you would like to have Starknet Foundry installed. A folder named snfoundry in
-   your [`%LOCALAPPDATA%\Programs`](https://learn.microsoft.com/en-us/windows/win32/shell/knownfolderid?redirectedfrom=MSDN#FOLDERID_UserProgramFiles)
-   directory will suffice:
+Verify if rust version >= 1.80.1 is installed
 
-```batch
-%LOCALAPPDATA%\Programs\snfoundry
+```shell
+$ rustc --version
+1.80.1
```
Comment (franciszekjob) [inline docs/src/getting-started/installation.md:237]: `$` prompt is on Unix-like os, not on Windows 😅  applies to other occurrences

**Review by franciszekjob — CHANGES_REQUESTED** [review]

File: docs/src/getting-started/installation.md [hunk 9]
```diff
@@ -1,103 +1,310 @@
...
+```shell
+rustup override set stable
+```
+
+#### Windows
+
+Follow [Rust installation](https://www.rust-lang.org/tools/install) and ensure correct version of rust was added to
+PATH.
+
+### `scarb test` Isn’t Running Tests Correctly
```
Comment (franciszekjob) [inline docs/src/getting-started/installation.md:289]: What does `Running Tests Correctly` mean? Maybe let's rephrase it to e.g. `Scarb isn't using snforge` or `Scarb isn't running tests with snforge`

File: docs/src/getting-started/installation.md [hunk 10]
```diff
@@ -1,103 +1,310 @@
...
+
+### No Version Set (Linux and MacOS Only)
 
 Users may encounter this error when trying to use `snforge` or `sncast` without setting a version:
 
 ```shell
 No version is set for command snforge
-Consider adding one of the following versions in your config file at starknet-foundry 0.32.0
+Consider adding one of the following versions in your config file at $HOME/.tool_versions
+starknet-foundry 0.32.0
```
Comment (franciszekjob) [inline docs/src/getting-started/installation.md:238]: nit: Will the version here be somehow swapped for latest one? It's the first place which user visits while interacting with foundry, so I think we should try to have the newest version here.

Comment (cptartur) [reply docs/src/getting-started/installation.md:238]: It's just an example error

**Review by franciszekjob — APPROVED** [review]
